suite-native/module-onboarding/src/screens/OnboardingXPub.tsx

```diff
@@ -0,0 +1,153 @@
+
+import { useFocusEffect } from '@react-navigation/native';
+
+import { Screen, StackProps } from '@suite-native/navigation';
+import { Box, Chip, Input, InputWrapper, Text } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { CryptoIcon } from '@trezor/icons';
+import { NetworkKey } from '@suite-common/wallet-config';
+
+import { Camera } from '../components/Camera';
+import { OnboardingStackParamList, OnboardingStackRoutes } from '../navigation/routes';
+
+const xpubContainerStyle = prepareNativeStyle(utils => ({
+    padding: utils.spacings.medium,
+}));
+
+const coinStyle = prepareNativeStyle(_ => ({
+    flexDirection: 'row',
+}));
+
+const cameraStyle = prepareNativeStyle(_ => ({
+    marginTop: 20,
+    marginBottom: 45,
+}));
+
+const cameraPlaceholderStyle = prepareNativeStyle(utils => ({
+    height: 329,
+    borderRadius: utils.borders.radii.medium,
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: utils.colors.gray800,
+}));
+
+const chipStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    borderRadius: utils.borders.radii.small,
+}));
+
+const DEFAULT_XPUB_INPUT_TEXT = '';
+const DEFAULT_CURRENCY_SYMBOL = 'btc';
+const DEFAULT_XPUB_RESULT = null;
+
+export const OnboardingXPub = ({
+    navigation,
+}: StackProps<OnboardingStackParamList, OnboardingStackRoutes.OnboardingXPub>) => {
+    const [xpubResult, setXpubResult] = useState<string | null>(DEFAULT_XPUB_RESULT);
+    const [selectedCurrencySymbol, setSelectedCurrencySymbol] =
+        useState<NetworkKey>(DEFAULT_CURRENCY_SYMBOL);
+    const [inputText, setInputText] = useState<string>(DEFAULT_XPUB_INPUT_TEXT);
+    const [cameraRequested, setCameraRequested] = useState<boolean>(false);
+    const { applyStyle } = useNativeStyles();
+
+    useFocusEffect(
+        useCallback(() => {
+            const setDefaultValues = () => {
+                setXpubResult(DEFAULT_XPUB_RESULT);
+                setSelectedCurrencySymbol(DEFAULT_CURRENCY_SYMBOL);
+                setInputText(DEFAULT_XPUB_INPUT_TEXT);
+                setCameraRequested(false);
+            };
+
+            setDefaultValues();
+        }, []),
+    );
+
+    useEffect(() => {
+        if (xpubResult) {
+            navigation.navigate(OnboardingStackRoutes.OnboardingAssets, {
+                xpubAddress: xpubResult,
+                currencySymbol: selectedCurrencySymbol,
+            });
+        }
+    }, [xpubResult, selectedCurrencySymbol, navigation]);
+
+    const handleSelectCurrency = (currencySymbol: NetworkKey) => {
+        setSelectedCurrencySymbol(currencySymbol);
+    };
+
+    const handleRequestCamera = () => {
+        setCameraRequested(true);
+    };
+
+    const handleXpubResult = (value?: string) => {
+        if (value) {
+            // FIXME: add isAddressValid validation here when @emurgo/cardano-serialization-lib-browser error will be resolved
```

Comment:
This was the problem when I tried to use `isAddressValid` from _@suite-common/wallet-utils_. In the end, I am letting xpub address be validated by trezor connect getAccountInfo and processing its response. But yes, this was the problem and we definitely should talk later about it. We can call tomorrow and I will describe this more.

Reply:
Please add this description of this to code as comment. Also please create follow-up task to fix solve this issue in because we should not have code in common that is not working for both platforms.